### PR TITLE
[DOCS] Adds a compound query example to the ELSER semantic search tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds an example of a compound query to the ELSER tutorial.

### Preview

[Combining `text_expansion` with a compound query](https://elasticsearch_96460.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-elser.html#text-expansion-compound-query)

### CODE REVIEW

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 1]
```diff
@@ -216,6 +216,57 @@ weights.
...
 // NOTCONSOLE
 
+
+[discrete]
+[[text-expansion-compound-query]]
+=== Combining `text_expansion` with a compound query
+
+You can combine `text_expansion` with a <<compound-queries,compound query>> for 
+example with a `query_string` that uses the same query text as the 
+`text_expansion` query. This enables you to combine the search results
```
Comment (davidkyle) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:226]: Can we expand on this with something like

```
You can combine `text_expansion` with other queries in a <<compound-queries,compound query>>. For 
example using a filter clause in a <<bool query link???https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html >> 
or a full text query which may or may not use the same query text as the 
`text_expansion` query.  This enables you to combine the search results from both queries.

The search hits from the `text_expansion` query tend to score higher than other Elasticsearch
 queries, those scores can be equalised/regularised by increasing or decreasing the relevance scores of each
query by using the `boost` parameter. Recall on the `text_expansion` query can be high where there is
a long tail of less relevant results, use the `min_score` parameter to prune those less relevant documents.

```

**Review by qherreros — APPROVED** [review]
LGTM !

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 2]
```diff
@@ -216,6 +216,69 @@ weights.
...
+
+[discrete]
+[[text-expansion-compound-query]]
+=== Combining `text_expansion` with a compound query
+
+You can combine `text_expansion` with a <<compound-queries,compound query>> for 
+example with a `query_string` that uses the same query text as the 
+`text_expansion` query. This enables you to combine the search results 
+from both queries. You can increase or decrease the relevance scores of each 
+query by using the `boost` parameter.
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:228]: With the next two paragraphs added in https://github.com/elastic/elasticsearch/pull/96460/commits/88118b99421749a5b2f4ba8e579e9fd3b1a650b5, this paragraph and the next have become repetitive. Maybe we drop this paragraph?

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 3]
```diff
@@ -216,6 +216,69 @@ weights.
 ----
 // NOTCONSOLE
 
+
+[discrete]
+[[text-expansion-compound-query]]
+=== Combining `text_expansion` with a compound query
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:222]:
```suggestion
=== Combining semantic search with other queries
```

Comment (abdonpijpelink) [reply docs/reference/search/search-your-data/semantic-search-elser.asciidoc:222]: Just a suggestion, feel free to ignore - I think this title would capture the "why" a bit better.

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 4]
```diff
@@ -216,6 +216,69 @@ weights.
...
+query by using the `boost` parameter.
+
+You can combine `text_expansion` with other queries in a 
+<<compound-queries,compound query>>. For example using a filter clause in a 
+<<query-dsl-bool-query>> or a full text query which may or may not use the same 
+query text as the `text_expansion` query. This enables you to combine the search 
+results from both queries.
+
+The search hits from the `text_expansion` query tend to score higher than other 
+{es} queries, those scores can be regularized by increasing or decreasing the
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:237]:
```suggestion
{es} queries. Those scores can be regularized by increasing or decreasing the 
```

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 5]
```diff
@@ -216,6 +216,69 @@ weights.
...
+<<compound-queries,compound query>>. For example using a filter clause in a 
+<<query-dsl-bool-query>> or a full text query which may or may not use the same 
+query text as the `text_expansion` query. This enables you to combine the search 
+results from both queries.
+
+The search hits from the `text_expansion` query tend to score higher than other 
+{es} queries, those scores can be regularized by increasing or decreasing the 
+relevance scores of each query by using the `boost` parameter. Recall on the 
+`text_expansion` query can be high where there is a long tail of less relevant 
+results, use the `min_score` parameter to prune those less relevant documents.
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:240]:
```suggestion
results. Use the `min_score` parameter to prune those less relevant documents.
```

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I've left a few small comments and suggestions.

**Pull request merged**